src/olympia/versions/models.py

```diff
@@ -142,8 +142,8 @@ def from_upload(cls, upload, addon, platforms, channel, send_signal=True,
             parsed_data = utils.parse_addon(upload, addon)
         license_id = None
         if channel == amo.RELEASE_CHANNEL_LISTED:
-            previous_version = addon.find_latest_version_including_rejected(
-                channel=channel)
+            previous_version = addon.find_latest_version(
+                channel=channel, exclude=())
```

Comment:
mmm I forgot about that, but I don't see a good reason why beta versions shouldn't have a license - they are released to end users and as such I feel a license is needed.

In addition this whole PR is based on the fact that beta versions do have a license, so they can be considered by `get_required_metadata()` when looking for if the add-on has a license or not...